stdlib/asm/math/gfp5.masm

```diff
@@ -0,0 +1,558 @@
+# Given two GF(p^5) elements on stack, this routine computes modular
+# addition over extension field GF(p^5) s.t. p = 2^64 - 2^32 + 1
+#
+# Expected stack state :
+#
+# [a0, a1, a2, a3, a4, b0, b1, b2, b3, b4, ...]
+#
+# After application of routine stack :
+#
+# [c0, c1, c2, c3, c4, ...]
+#
+# See section 3.2 of https://eprint.iacr.org/2022/274.pdf
+#
+# For reference implementation in high level language, see 
+# https://github.com/pornin/ecgfp5/blob/ce059c6/python/ecGFp5.py#L607-L616
+export.add
+    movup.5
+    add
+
+    swap
+    movup.5
+    add
+    swap
+
+    movup.2
+    movup.5
+    add
+    movdn.2
+
+    movup.3
+    movup.5
+    add
+    movdn.3
+
+    movup.4
+    movup.5
+    add
+    movdn.4
```

Comment:
I think we might be able to do it in 14 cycles like so:
```
repeat.5
  movup.5
  add
  movdn.4  
end
```

Reply:
Thanks for catching this @bobbinth .

I was also expecting cost of addition/ subtraction to decrease by 5 -cycles ( because now we're avoiding 5 stack movements ), but it turns out that cost only decreases by 3 -cycles. See commit message of https://github.com/maticnetwork/miden/pull/414/commits/eb4477a24ff80f65157bdb6c2656b52d10c16e96